### PR TITLE
fix: minor issues NTRN-402

### PR DESCRIPTION
**Pull request by sotnikov-s**
**task:** https://p2pvalidator.atlassian.net/browse/NTRN-402

subtask report:
1. was done in https://github.com/neutron-org/neutron-dao/pull/35;
2. was done in https://github.com/neutron-org/neutron-dao/pull/35;
3. make `owner` mandatory for the mentioned contracts + for the lockdrop and credits vaults;
4. done as suggested;
5. done as suggested;
6. don't get what auditors mean (`checked_div` doesn't return `OverflowError` but `DivideByZeroError`), so I've added a non-zero check and prettified errors handling;
7. removed the unused structs;
8. comment is removed since the feature is not going to be implemented (discussed with @oopcode);
9. idk what kind of validation `denom` and `min_period` require and what's wrong with the `vesting_denominator` validation, so I kept the original validation but moved it to a `Config` struct method and added tests for it. If you have an idea please let me know.

**relative PRs:**
- https://github.com/neutron-org/neutron/pull/166
- https://github.com/neutron-org/neutron-integration-tests/pull/104

**tests run:** https://github.com/neutron-org/neutron-tests/actions/runs/4446787903

### CODE REVIEW

File: contracts/dao/voting/credits-vault/src/contract.rs [hunk 1]
```diff
@@ -95,32 +84,25 @@ pub fn execute_update_config(
     deps: DepsMut,
     info: MessageInfo,
     new_credits_contract_address: Option<String>,
-    new_owner: Option<Admin>,
+    new_owner: String,
```
Comment (oopcode) [inline contracts/dao/voting/credits-vault/src/contract.rs:87]: why did you change `Admin` to `String`?

Comment (sotnikov-s) [reply contracts/dao/voting/credits-vault/src/contract.rs:87]: long story short: so it's **simpler** and **common**

**simpler:** because it's anyway eventually stored as `Addr` in the contract's cfg. just look a bit below at how this value was handled and how it's now handled. `Admin` struct is very complicated to be passed, just look at the changes here: https://github.com/neutron-org/neutron/pull/166/files. if we pass `Admin::CoreModule`, it simply sets the owner to the sender, so why don't allow sending the simple sender address?

**common:** another reason is that we have the owner field as String in all other contracts

Comment (oopcode) [reply contracts/dao/voting/credits-vault/src/contract.rs:87]: okay, i prefer to trust you!

**Review by oopcode — APPROVED** [review]

File: contracts/tokenomics/treasury/src/error.rs [hunk 2]
```diff
@@ -16,11 +16,11 @@ pub enum ContractError {
...
     NoFundsToDistribute {},
 
-    #[error("Invalid distribution rate")]
-    InvalidDistributionRate(String),
+    #[error("distribution_rate must be between 0 and 1")]
+    InvalidDistributionRate {},
 
-    #[error("Invalid vesting denominator")]
-    InvalidVestingDenominator(String),
+    #[error("vesting_denominator must be more than zero")]
```
Comment (NeverHappened) [inline contracts/tokenomics/treasury/src/error.rs:22]: more -> greater

File: contracts/tokenomics/treasury/src/state.rs [hunk 3]
```diff
@@ -21,10 +22,23 @@ pub struct Config {
     /// Address of the security DAO contract
     pub security_dao_address: Addr,
 
-    // Denomintator used int the vesting release function
+    // Denomintator used in the vesting release function
```
Comment (NeverHappened) [inline contracts/tokenomics/treasury/src/state.rs:25]: Denomintator -> Denominator

File: contracts/tokenomics/treasury/src/state.rs [hunk 4]
```diff
@@ -21,10 +22,23 @@ pub struct Config {
...
+impl Config {
+    pub fn validate(&self) -> Result<(), ContractError> {
+        if (self.distribution_rate > Decimal::one()) || (self.distribution_rate < Decimal::zero()) {
+            return Err(ContractError::InvalidDistributionRate {});
+        }
+
+        if self.vesting_denominator == 0 {
+            return Err(ContractError::InvalidVestingDenominator {});
+        }
+        Ok(())
```
Comment (NeverHappened) [inline contracts/tokenomics/treasury/src/state.rs:38]: maybe add check for `min_period > 0`?

Comment (NeverHappened) [reply contracts/tokenomics/treasury/src/state.rs:38]: in the task it also says to validate denom, but I'm not sure how

**Review by NeverHappened — CHANGES_REQUESTED** [review]

File: contracts/tokenomics/treasury/src/state.rs (hunk 4 above)
Comment (sotnikov-s) [reply contracts/tokenomics/treasury/src/state.rs:38]: neither do I:

> idk what kind of validation denom and min_period require and what's wrong with the vesting_denominator validation, so I kept the original validation but moved it to a Config struct method and added tests for it. If you have an idea please let me know.

yeah, min_period > 0 seems to be the only thing we can validate

**Review by NeverHappened — APPROVED** [review]
LGTM
